tests/mutex.rs

```diff
@@ -80,3 +80,16 @@ fn lifetime() {
         mutex.lock_arc()
     };
 }
+
+#[test]
+fn yields_when_contended() {
+    future::block_on(async {
+        let m = Mutex::new(());
+        let _g = m.try_lock().unwrap();
+        assert!(future::poll_once(m.lock()).await.is_none());
```

Comment:
This test should also make sure that the future wakes back up once the other lock is dropped. Ditto for all of the others.

Reply:
Does it suffice to check that the future now returns `Ready` or do you want to check that the `Waker` was triggered? I can do the latter but I think I'd need to add `waker-fn` as a dev dependency to do it cleanly.

Reply:
Doing both would be nice. It's fine to add `waker-fn` as a dev dependency, since it's already brought in via `futures-lite`.